### ISSUE TITLE
Google Provider v5.0 released, but there is version constraint `< 5.0.0`

### ISSUE THREAD

**Issue by bigwheel** [terraform-google-modules/terraform-google-github-actions-runners#123]
### TL;DR

[Google provider v5.0](https://registry.terraform.io/providers/hashicorp/google/5.0.0) was released few days ago, but the modules in this repo don't support v5.

### Additional information

By [Terraform Google Provider 5\.0\.0 Upgrade Guide \| Guides \| hashicorp/google \| Terraform \| Terraform Registry](https://registry.terraform.io/providers/hashicorp/google/latest/docs/guides/version_5_upgrade#provider), it seems like label behavior change may affect our modules.

version constraint lines:

- https://github.com/terraform-google-modules/terraform-google-github-actions-runners/blob/master/modules/gh-oidc/versions.tf#L23-L27
- https://github.com/terraform-google-modules/terraform-google-github-actions-runners/blob/master/modules/gh-runner-gke/versions.tf#L23
- https://github.com/terraform-google-modules/terraform-google-github-actions-runners/blob/master/modules/gh-runner-mig-vm/versions.tf#L23-L27
- https://github.com/terraform-google-modules/terraform-google-github-actions-runners/blob/master/modules/gh-runner-mig-container-vm/versions.tf#L23

**Comment by nikaro** [comment]
Would be fixed by https://github.com/terraform-google-modules/terraform-google-github-actions-runners/pull/124

**Comment by jmosborn** [comment]
I see this module has upgraded to `>= 6`, which is good, but using the module with v5 of the provider is still broken because of this version constraint: https://github.com/terraform-google-modules/terraform-google-github-actions-runners/blob/master/modules/gh-runner-mig-vm/main.tf#L122.

Google modules that reference other google modules should probably use `>=`. In this case it would fix the issue, as the instance_template module itself has upgrade to support v5 in its [latest version](https://github.com/terraform-google-modules/terraform-google-vm/blob/v10.1.1/modules/instance_template/versions.tf#L22).

**Comment by pickmylight** [comment]
I don't get it, why this is marked as stale? This is still an issue today. We stumbled upon this problem with the mig & mag-template dependency and had to downgrade our complete googleProvider to <5.0.0. What's the Point sticking with mig ~>7.0? There were 4 major updates since (11.x.x)
